src/main/java/org/opensearch/knn/jni/JNICommons.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.jni;
+
+import org.opensearch.knn.common.KNNConstants;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+/**
+ * Common class for providing the JNI related functionality to various JNIServices.
+ */
+public class JNICommons {
+
+    static {
```

Comment:
Do we need to explicitly ensure this gets called before faiss and nmslib get called?

Reply:
I didn't find any issue while running the cluster.

Reply:
and as of now there is no dependency between native libs and the commons. So we should be good for now.

Reply:
[This](https://github.com/navneet1v/k-NN/blob/stream-vectors/jni/src/org_opensearch_knn_jni_JNICommons.cpp#L21-L37) wont get called if one of the native libraries first calls it correct?

> and as of now there is no dependency between native libs and the commons. So we should be good for now.

I dont understand. https://github.com/navneet1v/k-NN/blob/stream-vectors/jni/include/jni_util.h is depended upon by both faiss and nmslib, right?

Reply:
So I validated that if Faiss Library gets loaded first too, then also we don't see any issue. Here are the steps I did.

1. First I added the log messages when Faiss and Commons lib were getting loaded.
2. I created a faiss index and did the indexing. This triggered Commons to get loaded first and then the faiss.
3. After this I killed the cluster and ran the cluster again with `--preserve-data` flag. 
4. After this I ran a search query on the index created with faiss engine. The search was successful. Only Faiss got loaded this time.
5. I ran the indexing again and this time commons got loaded. 
6. Did the search again, the search was successful.

Logs for step 3 and 4

```
[2024-04-10T13:43:09,162][INFO ][stdout                   ] [integTest-0] Loading Faiss library.....
[2024-04-10T13:43:09,174][WARN ][stderr                   ] [integTest-0] SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
[2024-04-10T13:43:09,175][WARN ][stderr                   ] [integTest-0] SLF4J: Defaulting to no-operation (NOP) logger implementation
[2024-04-10T13:43:09,175][WARN ][stderr                   ] [integTest-0] SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[2024-04-10T13:44:03,171][INFO ][stdout                   ] [integTest-0] Loading JNICommons library.....

```

Reply:
From java side I also checked if loadLibrary is called multiple times then there is no issues.

Doc: https://docs.oracle.com/javase/7/docs/api/java/lang/Runtime.html#loadLibrary%28java.lang.String%29:~:text=If%20this%20method%20is%20called%20more%20than%20once%20with%20the%20same%20library%20name%2C%20the%20second%20and%20subsequent%20calls%20are%20ignored.